feat(#599): Add Analytics Component to Admin.jsx

theboyofdream opened the pull request:
Issue no: #599 Add Analytics Card to Admin.jsx (i.e. /admin route)

## Description
- Added Analytics component to Admin.jsx
- made Analytics/AnalyticsCard responsive & arranged into single row using css flexbox
- wrote test-cases for
  - Analytics.jsx
  - AnalyticsCard.jsx
  - Admin.jsx
 
1st revision
- refactored css class names in Analytics.module.css & AnalyticsCard.module.css
- rearranged imports
- removed padding from Catalog.module.css file of .catalog-wrapper
- added test-id for test-cases & made some minor improvements
- removed /analytics route from app.jsx

2nd revision
- removed extra media-query padding from catalog.module.css

Changes other than the onces mention in issue
- fix label & text overlap in search input

## What type of PR is this? (Check all applicable)

- [x] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📄 Documentation Update
- [x] 👨‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [x] ✅ Test
- [ ] 🛠️ CI/CD

## Screenshots (if applicable)

https://github.com/user-attachments/assets/4fbb9851-daba-4a65-b58e-7b3ffdf83b2b

![Screenshot 2025-03-24 224708](https://github.com/user-attachments/assets/4305c54c-bed6-4e0d-9153-cdecb371d448)

![image](https://github.com/user-attachments/assets/125fd621-9b7f-4ca1-9dce-c0abd392c8a6)


## Checklist
- [x] I have performed a self-review of my code  
- [ ] I have commented my code, particularly in hard-to-understand areas  
- [x] I have added tests that prove my fix is effective or that my feature works  
- [x] New and existing unit tests pass locally with my changes

theboyofdream committed:
feat(#599): Add Analytics Component to Admin.jsx

- Added Analytics component to Admin.jsx
- made Analytics/AnalyticsCard them responsive & arranged into single row using css flexbox
- wrote test-cases for
  - Analytics.jsx
  - AnalyticsCard.jsx
  - Admin.jsx

theboyofdream committed:
refactor: removed commented code so sonarQubeCloud check will pass

mridxl commented:
Please add a screen recording of the changes and link your issue to the pr using the correct keywords.

aps08 commented:
> Please add a screen recording of the changes and link your issue to the pr using the correct keywords.

This is added just above the first screenshot.

aps08 commented on packages/ui/__tests__/components/analytics/Analytics.test.jsx:
```diff
@@ -0,0 +1,16 @@
+import { expect, describe, it } from "vitest";
+import { render, screen } from "@testing-library/react";
+import Analytics from "../../../src/components/analytics/Analytics";
+import { INFO } from "../../../src/utils/Constants";
```
`INFO` is very generic. Change the constant name

aps08 commented on packages/ui/__tests__/components/analytics/AnalyticsCard.test.jsx:
```diff
@@ -0,0 +1,20 @@
+
+    render(<AnalyticsCard title={info.title} api={info.api} />);
+
+    const apiText = info.api > 0 ? `+${info.api}%` : `-${info.api}%`;
```
what is this ?

aps08 commented on packages/ui/src/page/admin/Admin.jsx:
```diff
@@ -1,7 +1,13 @@
 import Catalog from "../../components/admindashboard/Catalog";
+import Analytics from "../../components/analytics/Analytics";
 
 function AdminDashboard() {
```
top and bottom padding is missing. Please check other public pages.

aps08 commented on packages/ui/__tests__/page/admin/Admin.test.jsx:
what is this  ?

aps08 commented on packages/ui/__tests__/components/analytics/Analytics.test.jsx:
```diff
@@ -0,0 +1,16 @@
+
+    INFO.forEach((item) => {
+      const titleElement = screen.getByText(item.title);
+
```
don't add unnecessary empty line. Check other test files.

aps08 commented on packages/ui/__tests__/components/analytics/AnalyticsCard.test.jsx:
```diff
@@ -0,0 +1,20 @@
+import AnalyticsCard from "../../../src/components/analytics/AnalyticsCard";
+import { INFO } from "../../../src/utils/Constants";
+import "@testing-library/jest-dom";
+import { expect, describe, it } from "vitest";
```
check import order in other files.

aps08 commented on packages/ui/src/components/analytics/Analytics.module.css:
correct the class naming standards. Please check header, footer and other public pages.

aps08 commented on packages/ui/src/components/analytics/AnalyticsCard.module.css:
correct the class naming standards. Please check header, footer and other public pages.

aps08 reviewed (changes requested):
Added comments

aps08 committed:
Merge branch 'stage' into enhancement/599-add-analytics-card-to-catalog-jsx-file

aps08 reviewed (changes requested):
Some changes were made to Dashboard page. Please take the latest pull and see if everything is working fine.

theboyofdream committed:
- refactored css class names in Analytics.module.css & AnalyticsCard.module.css
- rearranged imports
- removed padding from Catalog.module.css file of .catalog-wrapper
- fix label & text overlap in search input
- added test-id for test-cases & made some minor improvements
- removed /analytics route from app.jsx

theboyofdream committed:
- removed extra media-query padding from catalog.module.css

DaScientist commented on packages/ui/src/components/admindashboard/Catalog.jsx:
```diff
@@ -60,7 +60,8 @@ function Catalog() {
           label="search"
-          onChange={(e) => handleSearchTermChange(e)}
+          value={searchTerm}
+          onChange={handleSearchTermChange}
```
Why this change?

theboyofdream replied:
to fix the overlap of label & text inside search input in catalog component

theboyofdream committed:
removed extra padding medi-query from catalog.module.css

DaScientist reviewed (approved):
Changes look good to me
